<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PowerMeterOptmizerWindow</class>
 <widget class="QMainWindow" name="PowerMeterOptmizerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Power Meter Optimizer</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="horizontalWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <item>
        <widget class="QWidget" name="temperature_widget" native="true">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="2" column="2">
           <widget class="QDoubleSpinBox" name="T_B_doubleSpinBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>20</pointsize>
              <kerning>true</kerning>
             </font>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label">
            <property name="minimumSize">
             <size>
              <width>50</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>20</pointsize>
             </font>
            </property>
            <property name="text">
             <string>T_A</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_2">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>20</pointsize>
             </font>
            </property>
            <property name="text">
             <string>T_B</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="T_A_doubleSpinBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>20</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QCheckBox" name="connected_checkBox">
            <property name="text">
             <string>connected</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="start_pushButton">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="save_data_checkBox">
            <property name="text">
             <string>save h5</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="verticalSpacing">
           <number>0</number>
          </property>
          <item row="0" column="6">
           <widget class="QProgressBar" name="heater_output_progressBar">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="value">
             <number>24</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="label_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>setpoint T</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QComboBox" name="control_input_comboBox">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <widget class="QDoubleSpinBox" name="manual_heater_output_doubleSpinBox">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>55</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="buttonSymbols">
             <enum>QAbstractSpinBox::NoButtons</enum>
            </property>
            <property name="decimals">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="2" column="5" colspan="2">
           <widget class="QComboBox" name="control_mode_comboBox"/>
          </item>
          <item row="3" column="5" colspan="2">
           <widget class="QComboBox" name="heater_range_comboBox"/>
          </item>
          <item row="4" column="5" colspan="2">
           <widget class="QDoubleSpinBox" name="ramp_rate_doubleSpinBox">
            <property name="prefix">
             <string>ramp_rate: </string>
            </property>
           </widget>
          </item>
          <item row="1" column="5" colspan="2">
           <widget class="QDoubleSpinBox" name="setpoint_T_doubleSpinBox"/>
          </item>
          <item row="2" column="0" colspan="3">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>control mode</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="3">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>heater range</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="3">
           <widget class="QCheckBox" name="ramp_on_checkBox">
            <property name="font">
             <font>
              <pointsize>8</pointsize>
              <weight>50</weight>
              <bold>false</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>ramp_on</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="3">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>output power</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="verticalWidget" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QCheckBox" name="is_std_stable_checkBox">
            <property name="font">
             <font>
              <kerning>false</kerning>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">QCheckBox:disabled:checked{color:black} QCheckBox:disabled:unchecked{color:black}</string>
            </property>
            <property name="text">
             <string>Temperature stabilized</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="std_stable_doubleSpinBox"/>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="std_doubleSpinBox"/>
          </item>
          <item>
           <widget class="QPushButton" name="is_awake_pushButton">
            <property name="text">
             <string>awake</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="set_history_start_pushButton">
            <property name="toolTip">
             <string>Clears events and marks current time as &lt;i&gt;history_start&lt;/i&gt; of a new history.</string>
            </property>
            <property name="text">
             <string>set history_start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="save_history_pushButton">
            <property name="toolTip">
             <string>Saves temperature data and events between &lt;i&gt;history_start&lt;/i&gt; and now.</string>
            </property>
            <property name="text">
             <string>save from history_start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="event_text_lineEdit">
            <property name="toolTip">
             <string>Events are markers in the history. On &lt;i&gt; add event&lt;/i&gt; the text in this box will be added.</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="add_event_pushButton">
            <property name="toolTip">
             <string>Adds an event to the history. Events are markers in the history. </string>
            </property>
            <property name="text">
             <string>add event</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clear_events_pushButton">
            <property name="text">
             <string>clear events</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="plot_widget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
